common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/store/LwM2MBootstrapConfigStoreTaskProvider.java

```diff
@@ -213,44 +222,48 @@ private void initAfterBootstrapDiscover(BootstrapDiscoverResponse response) {
 
 
     public List<BootstrapDownlinkRequest<? extends LwM2mResponse>> toRequests(BootstrapConfig bootstrapConfig,
-                                                                              ContentFormat contentFormat) {
+                                                                              ContentFormat contentFormat,
+                                                                              Integer bootstrapServerIdOld,
+                                                                              String endpoint) {
         List<BootstrapDownlinkRequest<? extends LwM2mResponse>> requests = new ArrayList<>();
         Set<String> pathsDelete = new HashSet<>();
         List<BootstrapDownlinkRequest<? extends LwM2mResponse>> requestsWrite = new ArrayList<>();
         boolean isBsServer = false;
         boolean isLwServer = false;
         /** Map<serverId, InstanceId> */
         Map<Integer, Integer> instances = new HashMap<>();
+        Integer bootstrapServerIdNew = null;
         // handle security
         int id = 0;
         for (BootstrapConfig.ServerSecurity security : new TreeMap<>(bootstrapConfig.security).values()) {
             if (security.bootstrapServer) {
-                requestsWrite.add(toWriteRequest(this.securityInstances.get(0), security, contentFormat));
+                requestsWrite.add(toWriteRequest(this.lwM2MBootstrapSessionClients.get(endpoint).getSecurityInstances().get(0), security, contentFormat));
```

Comment:
Extract variable that will hold value of  "this.lwM2MBootstrapSessionClients.get(endpoint).getSecurityInstances().get(0)"

Reply:
You are using it in at least 5 places

Reply:
1. .getSecurityInstances().get(0) - this is bootstrapInstanceId in ObjectId = 0 "LWM2M Security";
2. .getSecurityInstances().get(0) == .getSecurityInstances().get(shortIdBootstrap)
3. "Short Server ID": This Resource MUST be set when the Bootstrap-Server Resource has a value of 'false'.
     * The values ID:0 and ID:65535 values MUST NOT be used for identifying the LwM2M Server.